### ISSUE TITLE
`table_region` does not imply order to returned table list

### ISSUE THREAD

**Issue by ottohirr** [camelot-dev/camelot#334]
### TLDR; 
Specifying a specific order to table locations via table_regions will not result in a returned list of tables within those regions. The return list of tables seems to be simply listed from top to bottom of the pages. This is an unexpected behavior (in my perspective). It means that there is no method to specify the order of the returned list of tables.

### Situation
Often reports are written in columnar format, where text flow proceeds from top left column, down the left column, then jumps to the top right column, down the right column, then jumps to the next page.

### Assumed use method
Most likely the typical usage by someone wanting to extract tables from such a report, would be to extract the tables in a logical order.

It would likewise seem logical that creating a specific ordered list of `table_regions` would result in extraction of tables from those regions in the order that the regions were specified.

So specifying `table_regions` of the left column, then the right column, would result in a list of tables occurring, first in the left column, ordered top to bottom, then in the right column ordered top to bottom.

### Actual behavior
**This is not the case. Tables appear to be returned by an order from top to bottom, regardless from which region they were extracted. I consider this is an unexpected behavior.**

Likewise this means that one cannot extract a set of tables in a preferred order, by specifying the order of `table_regions`.

One would have to make consecutive calls to `read_pdf`, with different `table_regions`, for the same page. This results in a huge repeat of the whole mechanism of reading a page.

.. Otto

**Comment by ottohirr** [comment]
After digging into how `table_regions` is used, it is not best to "extend" the concept of its use to include an ordering of tables as I suggested in original post. See its use in `Lattice._generate_table_bbox()`.

An ordering can be placed on the tables by examination of each table's `Table.cols` and `Table.rows` without having to change Camelot's library, though it would be nice to have within the library.

I'm going to close this out.
